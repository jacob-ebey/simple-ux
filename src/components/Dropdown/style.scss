@import '../variables';
@import '../../global';

.dropdown {
  display: inline-block;
  cursor: pointer;

  &-label {
    display: inline;
    &, &>* {
      cursor: pointer;
    }
  }

  &-content>* {
    color: $color-light;
  }
  
  &-content {
    display: none;
    z-index: 99;
    flex-direction: column;
    position: absolute;
    background-color: $color-lessdark;
    padding: $content-padding / 2;

    &>* {
      margin: $content-padding / 2 0;
    }
  }

  &:hover, &:active, &-content:hover, &-content:active {
    .dropdown-content {
      display: flex;
    }
  }
}
