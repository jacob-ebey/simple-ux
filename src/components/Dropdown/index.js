import { h } from 'preact' // eslint-disable-line no-unused-vars

import { merge } from '../tools'

import './style.scss'

const Dropdown = ({ label, class: className, children, ...rest }) => (
  <a
    {...rest}
    class={merge('dropdown', className)}
  >
    <div class='dropdown-label'>
      {label}
    </div>
    <div class='dropdown-content'>
      {children}
    </div>
  </a>
)

export default Dropdown
