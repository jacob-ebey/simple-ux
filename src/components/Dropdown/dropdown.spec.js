/* eslint-env jest */
import { h } from 'preact' // eslint-disable-line no-unused-vars
import { shallow } from 'preact-render-spy'

import Dropdown from './'

describe('Dropdown', () => {
  describe('- snapshots -', () => {
    it('can render', () => {
      const wrapper = shallow(<Dropdown />)
      expect(wrapper).toMatchSnapshot()
    })

    it('can add custom class', () => {
      const wrapper = shallow(<Dropdown class='custom' />)
      expect(wrapper).toMatchSnapshot()
    })

    it('can render children', () => {
      const wrapper = shallow(
        <Dropdown>
          <p>Child 1</p>
          <p>Child 2</p>
        </Dropdown>
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('can render label', () => {
      const wrapper = shallow(<Dropdown label={<a>Label</a>} />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
