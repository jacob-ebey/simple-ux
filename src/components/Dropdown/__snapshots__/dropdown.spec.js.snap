// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown - snapshots - can add custom class 1`] = `
preact-render-spy (1 nodes)
-------
<a class="dropdown custom">
  <div class="dropdown-label"></div>
  <div class="dropdown-content"></div>
</a>

`;

exports[`Dropdown - snapshots - can render 1`] = `
preact-render-spy (1 nodes)
-------
<a class="dropdown">
  <div class="dropdown-label"></div>
  <div class="dropdown-content"></div>
</a>

`;

exports[`Dropdown - snapshots - can render children 1`] = `
preact-render-spy (1 nodes)
-------
<a class="dropdown">
  <div class="dropdown-label"></div>
  <div class="dropdown-content">
    <p>Child 1</p>
    <p>Child 2</p>
  </div>
</a>

`;

exports[`Dropdown - snapshots - can render label 1`] = `
preact-render-spy (1 nodes)
-------
<a class="dropdown">
  <div class="dropdown-label">
    <a>Label</a>
  </div>
  <div class="dropdown-content"></div>
</a>

`;
